### PR TITLE
2673 Datetime col  to numpy  array not writeable error

### LINKED ISSUE #2673: Python numpy library has "assignment destination is read-only" error
I was testing the `deephaven-ib` PR for using DHaaL (https://github.com/deephaven-examples/deephaven-ib/pull/63).  The `examle_beta_calc.py` caused the following exception.

```
r-Scheduler-Serial-1 | i.d.s.s.SessionState      | Internal Error '20d0722b-5014-4119-acbe-7e1a703f3b35' java.lang.RuntimeException: Error in Python interpreter:
Type: <class 'deephaven.dherror.DHError'>
Value: failed to create a numpy array for the column ReceiveTime : ValueError: assignment destination is read-only
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/deephaven/numpy.py", line 47, in column_to_numpy_array
    np_array[:] = np_long_array
ValueError: assignment destination is read-only

Line: 63
Namespace: column_to_numpy_array
File: /usr/local/lib/python3.10/dist-packages/deephaven/numpy.py
Traceback (most recent call last):
  File "<string>", line 142, in <module>
  File "/usr/local/lib/python3.10/dist-packages/deephaven/pandas.py", line 77, in to_pandas
  File "/usr/local/lib/python3.10/dist-packages/deephaven/pandas.py", line 37, in _column_to_series
  File "/usr/local/lib/python3.10/dist-packages/deephaven/numpy.py", line 63, in column_to_numpy_array

        at org.jpy.PyLib.executeCode(PyLib.java:-2)
        at org.jpy.PyObject.executeCode(PyObject.java:138)
        at io.deephaven.engine.util.PythonEvaluatorJpy.evalScript(PythonEvaluatorJpy.java:73)
        at io.deephaven.integrations.python.PythonDeephavenSession.lambda$evaluate$1(PythonDeephavenSession.java:186)
        at io.deephaven.util.locks.FunctionalLock.doLockedInterruptibly(FunctionalLock.java:49)
        at io.deephaven.integrations.python.PythonDeephavenSession.evaluate(PythonDeephavenSession.java:185)
        at io.deephaven.engine.util.AbstractScriptSession.evaluateScript(AbstractScriptSession.java:166)
        at io.deephaven.engine.util.DelegatingScriptSession.evaluateScript(DelegatingScriptSession.java:81)
        at io.deephaven.engine.util.ScriptSession.evaluateScript(ScriptSession.java:106)
        at io.deephaven.server.console.ConsoleServiceGrpcImpl.lambda$executeCommand$8(ConsoleServiceGrpcImpl.java:188)
        at io.deephaven.server.session.SessionState$ExportBuilder.lambda$submit$2(SessionState.java:1299)
        at io.deephaven.server.session.SessionState$ExportObject.doExport(SessionState.java:847)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at io.deephaven.server.runner.DeephavenApiServerModule$ThreadFactory.lambda$newThread$0(DeephavenApiServerModule.java:162)
        at java.lang.Thread.run(Thread.java:829)
```

### PR DESCRIPTION
**Pull request by jmao-denver**
Fixes #2673

### CODE REVIEW

File: py/server/deephaven/table.py [hunk 1]
```diff
@@ -1389,6 +1390,30 @@ def partition_by(self, by: Union[str, Sequence[str]], drop_keys: bool = False) -
         except Exception as e:
             raise DHError(e, "failed to create a partitioned table.") from e
 
+    def update_by(self, ops: Union[UpdateByClause, List[UpdateByClause]],
```
Comment (chipkent) [inline py/server/deephaven/table.py:1393]: I doubt you meant to include all of this update by stuff in the datetime numpy fix.

File: py/server/deephaven/table_factory.py [hunk 2]
```diff
@@ -38,11 +38,12 @@ def empty_table(size: int) -> Table:
         raise DHError(e, "failed to create an empty table.") from e
 
 
-def time_table(period: str, start_time: str = None) -> Table:
+def time_table(period: Union[str, int], start_time: str = None) -> Table:
```
Comment (chipkent) [inline py/server/deephaven/table_factory.py:41]: Does this work?  Just checking that the cross type case works.
```
time_table(123, '2022-07-10T12:01:02 NY')
```

File: py/server/tests/test_table_factory.py [hunk 3]
```diff
@@ -52,6 +52,10 @@ def test_time_table(self):
         self.assertTrue(t.is_refreshing)
         self.assertEqual("2021-11-06T13:21:00.000000000 NY", t.j_table.getColumnSource("Timestamp").get(0).toString())
 
+        t = time_table(1000000000)
+        self.assertEqual(1, len(t.columns))
+        self.assertTrue(t.is_refreshing)
```
Comment (chipkent) [inline py/server/tests/test_table_factory.py:57]: Make sure you test all 4 branches based on type.  The (int,str) branch is not tested.

**Review by chipkent — CHANGES_REQUESTED** [review]

File: py/server/deephaven/table.py (hunk 1 above)
Comment (jmao-denver) [reply py/server/deephaven/table.py]: absolutely right

**Review by chipkent — APPROVED** [review]

**Pull request merged**